### ISSUE TITLE
User opens popup, and gets this message

### ISSUE THREAD

**Issue by skaefer143** [coe-google-apps-support/jazz-chrome#6]
![image](https://user-images.githubusercontent.com/22039892/33571571-a29a703e-d8ed-11e7-8d1d-15a3cf01d576.png)

This might be involved with the disconnection issue.

**Comment by skaefer143** [comment]
After more testing, only happens **randomly** after a chat has been closed, either because all available sessions have disconnected, or I just reloaded the extension.

**Comment by j-rewerts** [comment]
If it's due to an extension reload, that wouldn't be too bad. 

Start a chat with LiveChat support and see if they can give us some insight and control over lost connections. Ideally, I'd like to be able to respond and force another connection attempt if we lose the connection.

**Comment by skaefer143** [comment]
This message triggers in the console when connection is lost: 
![image](https://user-images.githubusercontent.com/22039892/34131160-b7d23034-e407-11e7-9db0-634f9e70686e.png)

I've seen this a few times, and I have no idea what causes it. It seems random. Waiting a few minutes somehow solves this.

**Comment by skaefer143** [comment]
Interestingly, running LC_API.start_chat(); actually restarts everything, including the chat and the connection. Too bad it's not supported as of right now, but I will look into this more when I hear back from Livechat.

**Comment by skaefer143** [comment]
This probably is connected with the OpenDNS thing now, I think. I get this in the popup.js console occasionally: 
![image](https://user-images.githubusercontent.com/22039892/34219579-e00eb3b6-e56e-11e7-829e-94948a01e825.png)

The connection is lost, and I can't figure out how to get it back again. Waiting 20 minutes usually works.

**Comment by j-rewerts** [comment]
Very strange. Should we engage LiveChat with this?

**Comment by skaefer143** [comment]
Maybe. I was talking with a support person yesterday, and they pretty much told me that there could potentially be security rules that are blocking things from triggering (on both livechats end and the chrome extensions end), and that's really all they know. That's why our chrome extension doesn't work occasionally.

It's hard to talk to a developer, since I get an email only once a day from them that usually isn't too helpful. I still haven't heard back about the start_chat() method, but apparently they will get back to me today.

I can engage LiveChat, but I don't think anything useful will come out of it.

**Comment by j-rewerts** [comment]
I found a weird thing with this. I initially see I'm connected with an agent, then it switches to the "Connection lost" message. If I type in and send a message super quick, I actually get connected properly.

**Comment by skaefer143** [comment]
Yup I have experienced that before. Reloading the extension does nothing, right? I have no idea how to trigger that. Next time you get that, can you clear the cookies in the developer console, then try again and see if it works? I think that could be a fix.

**Comment by j-rewerts** [comment]
That worked. I wonder if we can look for this specific error, then remove all of LiveChat's cookies?

**Comment by skaefer143** [comment]
With a mutation observer, it's possible

**Comment by j-rewerts** [comment]
Still attempting to reproduce this issue.

**Comment by j-rewerts** [comment]
I've explored this a little more. Jakub from live chat has given me a list of all cookies:
incap_ses_ - our DDoS protection
__lc.visitor_id - unique visitor's ID
__livechat_lastvisit - checks visitor's last visit
lc_window_state - checks the chat window state (minimized/maximized)
notification[predicted_agent] - assign the visitor to the specific agent
autoinvite_callback - checks if a visitor gets the invitation
__livechat - remembers the info of the visitor entering your website, 
common_iwcs_0 - synchronises the communication between opened chat tabs,
main_window_timestamp - as above, 
main_window_timestamp_0 - as above,
notification[new_messages] - informs about the pending messages, 
recent_window - informs about the last page that the visitor was on, 
3rdparty - checks if the browser's settings will allow to work on not yet visited pages
message_text - if you are writing a message and you refresh the page, this text will be remembered.
SERVERID_L - haproxy - our load balancer

**Comment by j-rewerts** [comment]
I just figured out how to reproduce this.

As a visitor, start a chat.
As an agent, respond to the visitor.
As a visitor, close your session and wait for the connection to terminate.
The agent should see this.
As a visitor, reopen a chat.
Note that the agent DOES NOT click "Close this window". They still see the old chat with the visitor.
This should result in the "Connection lost. Please check your internet connection, then try again" message.

It may be worth noting that I'm passing these URL params through to the url: name=jared.rewerts@edmonton.ca&email=jared.rewerts@edmonton.ca.

**Comment by j-rewerts** [comment]
LiveChat has been unable to reproduce the issue, but I can still reproduce it. 
I've found that deleting LiveChat's cookies resolves the issue. If LiveChat can't reproduce the issue on their end, we can always use that as the nuclear option.

**Comment by j-rewerts** [comment]
Users are reporting this issue now.

**Comment by j-rewerts** [comment]
LiveChat can now reproduce. LiveChat is working on resolving it.
